monitoring/mock_uss/interaction_logging/interactions.py

```diff
@@ -0,0 +1,28 @@
+from enum import Enum
+from typing import List
+
+from implicitdict import ImplicitDict
+import yaml
+from yaml.representer import Representer
+
+from monitoring.monitorlib.fetch import Query
+
+
+class QueryDirection(str, Enum):
+    Incoming = "Incoming"
+    """The query originated from a remote client and was handled by the system reporting the interaction."""
+
+    Outgoing = "Outgoing"
+    """The system reporting the interaction initiated the query to a remote server."""
+
+
+class Interaction(ImplicitDict):
+    query: Query
+    direction: QueryDirection
+
```

Comment:
issues field has been removed.
Apart from data validation, we will be adding message signing validation as well in mock_uss. If issues field is there we can communicate back the message signing validation failure.
Response in query can tell if there was any issue. But the details won't be known. We also would need to do message signing validation of response. Lack of issues field would mean we would have to do message signing validation again in uss_qualifier. Would that duplication be ok?

Reply:
Yes, requirement compliance verification (in uss_qualifier) should not depend on logic outside uss_qualifier since that outside logic is not version-controlled with uss_qualifier.  Duplication of (nominally) the same logic for two different purposes (mock_uss behavior + uss_qualifier compliance verification) should be fine.